src/valkey.c

```diff
@@ -772,9 +772,9 @@ int valkeyReconnect(valkeyContext *c) {
     c->err = 0;
     memset(c->errstr, '\0', strlen(c->errstr));
 
-    if (c->funcs && c->funcs->close) {
+    assert(c->funcs);
```

Comment:
Maybe it's acceptable but I think this will still lead to UB if libvalkey is compiled with `-DNDEBUG`.

Reply:
Ah.. `valkeyContextSetFuncs()` should set c->funcs, but its assert is not checked then either I guess..
I'll re-add the check here, and add checks in all other c->funcs calls in valkeyReconnect.